pkg/kubelet/kubeletconfig/util/test/test.go

```diff
@@ -40,16 +42,11 @@ func ExpectError(t *testing.T, err error, substr string) {
 // and logs the appropriate error on the test object.
 // The return value indicates whether we should skip the rest of the test case due to the error result.
 func SkipRest(t *testing.T, desc string, err error, contains string) bool {
-	if err != nil {
-		if len(contains) == 0 {
-			t.Errorf("case %q, expect nil error but got %q", desc, err.Error())
-		} else if !strings.Contains(err.Error(), contains) {
-			t.Errorf("case %q, expect error to contain %q but got %q", desc, contains, err.Error())
-		}
-		return true
-	} else if len(contains) > 0 {
-		t.Errorf("case %q, expect error to contain %q but got nil error", desc, contains)
-		return true
+	// return bools are negated because we skip when asserts fail, not when they succeed
```

Comment:
I'd lean towards just leaving this function the same, for fear of breaking some part of its behavior (have you thought through all the truth tables between the two implementations here and ensured that nothing changes?).

Reply:
Yeah I might have been a bit overzealous with the refactoring here. This isn't really something we need to change right now; I've reverted the change.

Reply:
sg